analysis/load_program.go

```diff
@@ -39,27 +40,39 @@ const PkgLoadMode = packages.NeedName |
 	packages.NeedTypesSizes |
 	packages.NeedModule
 
+// LoadProgramOptions combines all the options that are used when loading programs.
+type LoadProgramOptions struct {
+	// BuildMode is the mode used when creating the SSA from the packages.
+	BuildMode ssa.BuilderMode
+	// LoadTests is a flag indicating whether tests should be loaded with the program.
+	LoadTests bool
+	// ApplyRewrites is a flag indicating whether the standard source rewrites should be applied.
+	ApplyRewrites bool
+	// Platform indicates which platform the analysis is being performed on (sets GOOS in env).
+	Platform string
```

Comment:
Should this be an enum (I guess the Go equivalent is constants with a custom type) to restrict the set of platforms to the ones Go supports? Maybe there's already one defined in the standard library or x/tools?

Reply:
Yes I can look into that.